### PR TITLE
Add getThelen2003MuscleList() & getMillard2012EquilibriumMuscleList()

### PR DESCRIPTION
**Pull request by klshrinidhi**
Addressing #643.

Reviewer -- @aymanhab, @tkuchida

### CODE REVIEW

File: Bindings/OpenSimHeaders_simulation.h [hunk 1]
```diff
@@ -138,5 +138,12 @@
 #include <OpenSim/Simulation/StatesTrajectory.h>
 #include <OpenSim/Simulation/StatesTrajectoryReporter.h>
 
+#include <OpenSim/Actuators/osimActuatorsDLL.h>
+#include <OpenSim/Actuators/MuscleFixedWidthPennationModel.h>
```
Comment (tkuchida) [inline Bindings/OpenSimHeaders_simulation.h]: Is it necessary to include these Actuator headers in OpenSimHeaders_simulation.h? Currently, each OpenSimHeaders_[blah].h file strictly includes headers from OpenSim/[blah]. Why is it now necessary to deviate from this pattern?

File: Bindings/simulation.i [hunk 2]
```diff
@@ -229,3 +229,24 @@
...
+%template(Thelen2003MuscleIterator)
+    OpenSim::ComponentListIterator<const OpenSim::Thelen2003Muscle>;
+
+%template(Millard2012EquilibriumMuscleList)
+    OpenSim::ComponentList<const OpenSim::Millard2012EquilibriumMuscle>;
+%template(Millard2012EquilibriumMuscleIterator)
+    OpenSim::ComponentListIterator<const OpenSim::Millard2012EquilibriumMuscle>;
+
+%template(getThelen2003MuscleList)
+  OpenSim::Model::getComponentList<Thelen2003Muscle>;
```
Comment (tkuchida) [inline Bindings/simulation.i]: Probably doesn't matter, but based on the rest of this file, `Thelen2003Muscle` should be `OpenSim::Thelen2003Muscle`.

File: Bindings/OpenSimHeaders_simulation.h (hunk 1 above)
Comment (klshrinidhi) [inline Bindings/OpenSimHeaders_simulation.h]: I think it is necessary to have them here to add these functions to `Model`.

Comment (tkuchida) [inline Bindings/OpenSimHeaders_simulation.h]: Ah, ok. Thanks.

**Comment by aymanhab** [comment]
@klshrinidhi Is there a reason ClutchedPathSpring header was removed from the actuators.i file or did it go somewhere else? Thanks.

**Comment by klshrinidhi** [comment]
>@klshrinidhi Is there a reason ClutchedPathSpring header was removed from the actuators.i file or did it go somewhere else? Thanks.

It is still in `actuators.i`. You can see it [here](
https://github.com/opensim-org/opensim-core/blob/issue_643_component_list_muscles/Bindings/actuators.i#L8). Not sure why GitHub doesn't show.

**Comment by aymanhab** [comment]
Thanks @klshrinidhi for the clearup.
One last item, did you consider having the Java test perform "safeDownCast" to check the type (similar to the python test) so that property names are not hard coded and the test can stay intact regardless of changes in property names?

**Comment by klshrinidhi** [comment]
> One last item, did you consider having the Java test perform "safeDownCast" to check the type (similar to the python test) so that property names are not hard coded and the test can stay intact regardless of changes in property names?

I have changed the test to look like the one in Python.

**Comment by aymanhab** [comment]
Thank you @klshrinidhi and @tkuchida

**Pull request merged**